### DISCUSSION TITLE
Basis function labels

### DISCUSSION THREAD

**Discussion by kvkarandashev** [tblite/tblite#49] [Ideas]
Hello all!

I was wondering whether it is possible to get AO labels (preferably in a string format similar to pySCF's) from tblite at this point; it would be great if this could be done inside the Python API along with all the other information about the calculation.

Thank you!

**Comment by awvwgk** [comment]
The `Calculator` can provide access to the AO information, currently only the mapping of the AOs to the shells and to the atoms centers is provided. Most of the basis set related information are stored in the [`basis_type`](https://github.com/tblite/tblite/blob/main/src/tblite/basis/type.f90) class.

---

For example the orbital mapping the getter is defined in Fortran at

https://github.com/tblite/tblite/blob/08f1bb94d933501d7dc2f94eda0e2a9c63b18df7/src/tblite/api/calculator.f90#L430-L454

Exported via C at

https://github.com/tblite/tblite/blob/08f1bb94d933501d7dc2f94eda0e2a9c63b18df7/include/tblite/calculator.h#L122-L126

Wrapped in the CFFI layer at

https://github.com/tblite/tblite/blob/08f1bb94d933501d7dc2f94eda0e2a9c63b18df7/python/tblite/library.py#L352-L360

And exposed as getter in

https://github.com/tblite/tblite/blob/08f1bb94d933501d7dc2f94eda0e2a9c63b18df7/python/tblite/interface.py#L416

New properties like the angular momenta can be added in a similar way.

**Comment by kvkarandashev** [comment]
Thank you! I think angular momentum is the last bit of information I would need; if there are no updates on this I will try to make the respective commit next weekend.

**Reply by awvwgk** [reply]
Sounds good to me.

**Comment by MoleOrbitalHybridAnalyst** [comment]
Thank both of you! I wonder if it is possible to get more detailed information about the basis, e.g. I want to know whether a p-orbital is px, py or pz. This would be greatly helpful for me to understand which is which in the density matrix I retrieve from the result.

**Reply by awvwgk** [reply]
The orbitals within a shell are order from -l to +l, this should allow you to infer which function is which.

**Reply by MoleOrbitalHybridAnalyst** [reply]
Thanks so much!
